src/connector/src/sink/iceberg/mod.rs

```diff
@@ -114,29 +118,17 @@ pub struct IcebergConfig {
     #[serde(
         rename = "primary_key",
         default,
-        deserialize_with = "deserialize_string_seq_from_string"
+        deserialize_with = "deserialize_optional_string_seq_from_string"
     )]
     pub primary_key: Option<Vec<String>>,
 
     // Props for java catalog props.
     #[serde(skip)]
     pub java_catalog_props: HashMap<String, String>,
-}
 
-pub(crate) fn deserialize_string_seq_from_string<'de, D>(
-    deserializer: D,
-) -> std::result::Result<Option<Vec<String>>, D::Error>
-where
-    D: de::Deserializer<'de>,
-{
-    let s: Option<String> = de::Deserialize::deserialize(deserializer)?;
-    if let Some(s) = s {
-        let s = s.to_ascii_lowercase();
-        let s = s.split(',').map(|s| s.trim().to_owned()).collect();
-        Ok(Some(s))
-    } else {
-        Ok(None)
-    }
+    // Commit every n checkpoints, if n is not set, 0, 1, we will commit every checkpoint.
```

Comment:
Any reason to specify this special deserialization method? What the difference to the default one?

Reply:
config actually will be deserialize from string. So we need to specify a method from str to other type. Like https://github.com/risingwavelabs/risingwave/blob/61c025d1c7d1705c193901e98c57b221c0f9a711/src/connector/src/sink/kafka.rs#L217